src/coreclr/jit/codegencommon.cpp

```diff
@@ -5399,6 +5399,17 @@ void CodeGen::genFinalizeFrame()
     noway_assert(!regSet.rsRegsModified(RBM_FPBASE));
 #endif
 
+#ifdef SWIFT_SUPPORT
+    // If this method returns an error value in the Swift error register,
+    // don't push/pop REG_SWIFT_ERROR.
+    if (compiler->lvaSwiftErrorArg != BAD_VAR_NUM)
+    {
+        assert(compiler->info.compCallConv == CorInfoCallConvExtension::Swift);
+        assert(compiler->lvaSwiftErrorLocal != BAD_VAR_NUM);
+        regSet.rsRemoveRegsModified(RBM_SWIFT_ERROR);
+    }
+#endif // SWIFT_SUPPORT
```

Comment:
It doesn't seem right that we mark this register as unmodified since we _are_ going to modify it. Rather, it seems like we should adjust the prolog generation code to not consider this register to be callee saved in these methods.